File: Data Structures & Algorithms/count-prefix-and-suffix-pairs-i/submission-0.py
class Solution:
    def isPrefixAndSuffix(self, str1, str2) -> bool:
        if str2.startswith(str1) and str2.endswith(str1):
            return True
        return False

    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        result = 0
        
        for i in range(len(words)):
            for j in range(i+1, len(words)):
                first = words[i]
                second = words[j]
                if self.isPrefixAndSuffix(first, second):
                    print('are you here?', first, second)
                    result += 1

        return result

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> words;

        for (int i = 0; i < strs.size(); ++i) {
            string s = strs[i];
            sort(s.begin(), s.end());
            words[s].push_back(strs[i]);
        }

        vector<vector<string>> result;
        for (auto& word : words) {
            result.push_back(word.second);
        }

        return result;
    }
};
